twisted/test/test_twistd.py

```diff
@@ -1802,7 +1857,7 @@ def startService(self):
         self.assertEqual(
             self.mockos.actions,
             [('chdir', '.'), ('umask', 0o077), ('fork', True), 'setsid',
-             ('fork', True), ('write', -2, '1 ' + 'x' * 98),
+             ('fork', True), ('write', -2, b'1 ' + b'x' * 98),
```

Comment:
instead of `b'x'` maybe you can add `u'\N{SNOWMAN}'.encode('utf-8')[:-1]` to get some non-ascii bytes and see what it happens if we receive partial UTF-8 characters